src/streamlink/session.py

```diff
@@ -291,8 +291,9 @@ def set_option(self, key, value):
             self.http.cert = value
         elif key == "http-timeout":
             self.http.timeout = value
-        else:
-            self.options.set(key, value)
+
+        # all options are available via `options.get`
+        self.options.set(key, value)
```

Comment:
should probably be its own PR

---

`get_option` and `options.get` might be different _(not directly related to your changes)_ 

```
        print(self.session.get_option("http-proxy"))
http://127.0.0.1:80
        print(self.session.options.get("http-proxy"))
127.0.0.1:80
```

Reply:
True, it's not required when using `self.session.get_option` :)